subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -1278,36 +1277,15 @@ uint8_t ll_adv_enable(uint8_t enable)
 	lll->is_hdcd = !interval && (pdu_adv->type == PDU_ADV_TYPE_DIRECT_IND);
 	if (lll->is_hdcd) {
 		ret_cb = TICKER_STATUS_BUSY;
-#if defined(CONFIG_BT_TICKER_EXT)
-#if !defined(CONFIG_BT_CTLR_JIT_SCHEDULING)
-		ll_adv_ticker_ext[handle].ticks_slot_window = 0;
-#endif /* CONFIG_BT_CTLR_JIT_SCHEDULING */
-		ll_adv_ticker_ext[handle].ext_timeout_func = ticker_cb;
-		ll_adv_ticker_ext[handle].expire_info_id = TICKER_NULL;
-
-		ret = ticker_start_ext(
-#else
-		ret = ticker_start(
-#endif
-				   TICKER_INSTANCE_ID_CTLR,
+		ret = ticker_start(TICKER_INSTANCE_ID_CTLR,
 				   TICKER_USER_ID_THREAD,
 				   (TICKER_ID_ADV_BASE + handle),
 				   ticks_anchor, 0,
 				   (adv->ull.ticks_slot + ticks_slot_overhead),
 				   TICKER_NULL_REMAINDER, TICKER_NULL_LAZY,
 				   (adv->ull.ticks_slot + ticks_slot_overhead),
-#if defined(CONFIG_BT_TICKER_EXT)
-				   NULL,
-#else
-				   ticker_cb,
-#endif /* CONFIG_BT_TICKER_EXT */
-				   adv,
-				   ull_ticker_status_give, (void *)&ret_cb
-#if defined(CONFIG_BT_TICKER_EXT)
-				   ,
-				   &ll_adv_ticker_ext[handle]
-#endif /* CONFIG_BT_TICKER_EXT */
-				   );
+				   ticker_cb, adv,
+				   ull_ticker_status_give, (void *)&ret_cb);
```

Comment:
Since we use the same `ticker_cb` function here as in the other cases, you have to use the ticker_start_ext "version" if `CONFIG_BT_TICKER_EXT_EXPIRE_INFO` is defined (otherwise `param` in `ticker_cb` doesn't have the correct type, leading to a crash when using hdcd)

Reply:
Done